File: Java-Python-SQL-Fundamentals/Java Fundamentals/Inheritance-Practice3/src/main/java/org/example/ClientAccount.java
package org.example;

//CHILD
public class ClientAccount extends Account {
    private String membershipLevel;
    private ShoppingCart shoppingCart;


    public ClientAccount(String username, String email, String membershipLevel) {
        super(username, email);
        this.membershipLevel = membershipLevel;
        shoppingCart = new ShoppingCart("My Shopping Cart");

    }

    public void addItemToCart() {

        if (!isUserLoggedIn()) {
            System.out.println("Error: Login to Account\n");
            return;
        }

        shoppingCart.addItem();

    }

    public void removeItemFromCart() {
        if (!isUserLoggedIn()) {
            System.out.println("Error: Login to Account\n");
            return;
        }

        shoppingCart.removeItem();


    }

    public void clearShoppingCart() {
        if (!isUserLoggedIn()) {
            System.out.println("Error: Login to Account\n");
            return;
        }
        shoppingCart.clearCart();
    }

    public void upgradeMembership(String newLevel) {
        if (!isUserLoggedIn()) {
            System.out.println("Error: Login to Account\n");
            return;
        }

        if (membershipLevel.equals(newLevel)) {

            System.out.println("Already using " + membershipLevel);
        } else {
            System.out.println("Membership upgraded to " + newLevel + "\n");
            membershipLevel = newLevel;
        }


    }

    public void viewCart() {
        if (!isUserLoggedIn()) {
            System.out.println("Error: Login to Account\n");
            return;
        }

        System.out.println(shoppingCart.displayCart());

    }

    public String getAccDetails() {

        return getAccountInfo() + "Membership: " + membershipLevel + "\n";

    }

}

File: Java-Python-SQL-Fundamentals/Python-Fundamentals/Python_OOP_Inheritance2/Client_Account.py
from Account import MainAccount
from Shopping_cart import ShoppingCart


class ClientAccount(MainAccount):

    def __init__(self, username, email, membership_level, shopping_exs):

        super().__init__(username, email)
        self.membership_level = membership_level
        self.shopping_cart = ShoppingCart(shopping_exs)

    def add_to_cart(self, item):

        if not self._is_logged_in():
            print('Need to login first!')
            return

        self.shopping_cart.add_item(item)

    def remove_from_cart(self, item):

        if not self._is_logged_in():
            print('Need to login first!')
            return

        self.shopping_cart.remove_item(item)

    def empty_cart(self):

        if not self._is_logged_in():
            print('Need to login first!')
            return

        self.shopping_cart.clear_cart()

    def view_cart(self):

        if not self._is_logged_in():
            print('Need to login first!')
            return

        print(self.shopping_cart.display_cart())

    def upgrade_membership(self, new_level):

        if not self._is_logged_in():
            print('Need to login first!')
            return

        if self.membership_level == new_level:
            print(f'Already a {self.membership_level} member \n')
            return

        self.membership_level = new_level
        print(f'Successfully upgraded to {self.membership_level}')

    def get_client_details(self) -> str:

        return f'{self.get_account_info()} \nMembership Level: {self.membership_level} \n'
